jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java

```diff
@@ -586,21 +505,17 @@ public boolean onRemoveField(HttpField field)
                 return false;
             if (field.getHeader() == null)
                 return true;
-            switch (field.getHeader())
+            if (Objects.requireNonNull(field.getHeader()) == HttpHeader.CONTENT_TYPE)
```

Comment:
`Objects.requireNonNull()` is not necessary, as the null check is 2 lines above.

No more case for `CONTENT_LENGTH`?

Reply:
The content length recorded here is at the context level, which could be before or after compression, depending on configuration.  We can do all the required checks at HttpOutput level for application writes/completion and HttpChannel level for network writes/completion